crates/sui-core/src/checkpoints/checkpoint_output.rs

```diff
@@ -81,7 +81,11 @@ impl<T: SubmitToConsensus + ReconfigurationInitiator> CheckpointOutput
             .submit_to_consensus(&transaction, epoch_store)
             .await?;
         if let Some(checkpoints_per_epoch) = self.checkpoints_per_epoch {
-            if checkpoint_seq != 0 && checkpoint_seq % checkpoints_per_epoch == 0 {
+            let epoch = epoch_store.epoch();
+            if checkpoint_seq != 0
+                && ((epoch == 0 && checkpoint_seq % 28800 == 0)
```

Comment:
Maybe `(checkpoint_seq == 28800) || (checkpoint_seq > 28800 && (checkpoint_seq - 28800) % checkpoints_per_epoch == 0)`?

Reply:
Is there a concern to leave enough time for the upgrade before another epoch change? I'm fine either way and you can decide.

Reply:
Also do we want to hardcode in the new checkpoint per epoch value to ~20k here or do we want to rely on validators changing their config?

Reply:
I left it to the config so that we could change it latter if we need to without having to re-roll a binary